File: Python/Leetcode/String/intToRoman.py
class Solution(object):
    def intToRoman(self, num):
        s = ''
        map = {1000: 'M', 900 : 'CM', 500: 'D', 400: 'CD', 100: 'C', 90: 'XC', 50: 'L', 40: 'XL', 10: 'X', 9: 'IX', 5: 'V', 4:'IV', 1: 'I'}
        for val, numeral in sorted(map.items(), key=lambda x: x[0], reverse=True):
            while num >= val:
                num -= val
        return s
    
if __name__ == '__main__':
    solution_instance = Solution()
    solution_instance.intToRoman(3)

File: C++/Leetcode/Hashmap/intToRoman.cpp
#include <iostream>
#include <map>

class Solution {
    public:
        std::string intToRoman(int num){
            std::map<int, std::string, std::greater<int>> romanNumerals = {  // Adds the comparator
                {1000, "M"},
                {900, "CM"},
                {500, "D"},
                {400, "CD"},
                {100, "C"},
                {90, "XC"},
                {50, "L"},
                {40, "XL"},
                {10, "X"},
                {9, "IX"},
                {5, "V"},
                {4, "IV"},
                {1, "I"},
            };

            std::string res;

            for(std::map<int, std::string>::iterator itr = romanNumerals.begin(); itr != romanNumerals.end(); ++itr){
                while (num >= itr->first){
                    res += itr->second;
                    num -= itr->first;
                }
            }
            return res;
        }
};

int main(){
    Solution solution_instance;
    std::cout << solution_instance.intToRoman(1994) << std::endl;
    return 0;
}